src/vcpkg/binarycaching.cpp

```diff
@@ -95,20 +104,71 @@ namespace
                                  (msg::binary_source),
                                  "",
                                  "invalid argument: binary config '{binary_source}' expects a single string argument");
-    DECLARE_AND_REGISTER_MESSAGE(InvalidArgumentRequiresOneOrTwoArguments,
+    DECLARE_AND_REGISTER_MESSAGE(InvalidArgumentRequiresSource,
                                  (msg::binary_source),
                                  "",
-                                 "invalid argument: binary config '{binary_source}' requires 1 or 2 arguments");
+                                 "invalid argument: binary config '{binary_source}' requires non-empty source");
+    DECLARE_AND_REGISTER_MESSAGE(
+        InvalidArgumentRequiresSourceArgument,
+        (msg::binary_source),
+        "",
+        "invalid argument: binary config '{binary_source}' requires at least a source argument");
     DECLARE_AND_REGISTER_MESSAGE(InvalidArgumentRequiresTwoOrThreeArguments,
                                  (msg::binary_source),
                                  "",
                                  "invalid argument: binary config '{binary_source}' requires 2 or 3 arguments");
-    DECLARE_AND_REGISTER_MESSAGE(CompressFolderFailed, (msg::path), "", "Failed to compress folder '{path}':");
+    DECLARE_AND_REGISTER_MESSAGE(InvalidArgumentRequiresValidToken,
+                                 (msg::binary_source),
+                                 "",
+                                 "invalid argument: binary config '{binary_source}' requires a SAS token without a "
+                                 "preceeding '?' as the second argument");
+    DECLARE_AND_REGISTER_MESSAGE(PackingVendorFailed,
+                                 (msg::vendor),
+                                 "",
+                                 "Packing {vendor} failed.  Use --debug for more information.");
+    DECLARE_AND_REGISTER_MESSAGE(PushingVendorFailed,
+                                 (msg::vendor, msg::path),
+                                 "",
+                                 "Pushing {vendor} to '{path}' failed.  Use --debug for more information.");
+    DECLARE_AND_REGISTER_MESSAGE(ReplaceSecretsError, (msg::error_msg), "", "'{error_msg}'");
+    DECLARE_AND_REGISTER_MESSAGE(RestoredPackage, (msg::path), "", "Restored package from '{path}'");
+    DECLARE_AND_REGISTER_MESSAGE(
+        RestoredPackagesFromVendor,
+        (msg::count, msg::elapsed, msg::vendor),
+        "",
+        "Restored {count} package(s) from {vendor} in {elapsed}. Use --debug to see more details.");
```

Comment:
```suggestion
    DECLARE_AND_REGISTER_MESSAGE(
        RestoredPackagesFromSource,
        (msg::count, msg::elapsed, msg::value),
        "{value} may be either a 'vendor' like 'Azure' or 'NuGet', or a file path like C:\\example or /usr/example",
        "Restored {count} package(s) from {value} in {elapsed}. Use --debug to see more details.");
```

(And fix callers)

Now that you're passing file paths in here rather than only vendor names.

Reply:
I think it would be even better to create something like `msg::packagesource` which can be reused.